test/app-luatest/tap_test.lua

```diff
@@ -0,0 +1,633 @@
+-- The test for The Test Anything Protocol module.
+--
+-- Disable stack traces for this test because Tarantool test
+-- system also checks test output.
+
+local t = require('luatest')
+local treegen = require('luatest.treegen')
+local justrun = require('luatest.justrun')
+
+local g = t.group()
+
+-- The testcase tests `ok`, `fail` and `skip` predicates.
+g.test_tap_module_test_statuses = function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'tap_test_tap_module_test_statuses.lua'
+    treegen.write_file(dir, script_name, [[
+local tap = require('tap')
+local test = tap.test('root test')
+test.trace = false
+test:plan(5)
+test:ok(true, 'true') -- basic function
+local extra = { state = 'some useful information to debug on failure',
+        details = 'a table argument formatted using yaml.encode()' }
+test:ok(true, "extra information is not printed on success", extra)
+test:ok(false, "extra printed using yaml only on failure", extra)
+
+test:fail('failed') -- always fail the test
+test:skip('test marked as ok and skipped')
+    ]])
+    local opts = {nojson = true, stderr = true}
+    local res = justrun.tarantool(dir, {}, { script_name }, opts)
+    t.assert_equals(res.exit_code, 0)
+    t.assert_equals(res.stdout, [[
+TAP version 13
+1..5
```

Comment:
Side note: TBH, the huge TAP outputs look dramatically big. May be it is better to introduce a helper that will append the TAP preambule for a single test:

``` 
TAP version 13
1..1
```
And separate testcases 1 by 1 where it is possible. Feel free to ignore though.

Reply:
Updated test has a testcase for checking TAP output and a testcase for `is_deeply()`, both parameterized.

However, several testcases still have a huge TAP output, I'm afraid that checking a complex output by luatest assertions is risky and left these testcases as is.

```diff
diff --git a/test/app-luatest/tap_test.lua b/test/app-luatest/tap_test.lua
index 74d10d3c1f..dd194bb720 100644
--- a/test/app-luatest/tap_test.lua
+++ b/test/app-luatest/tap_test.lua
@@ -7,207 +7,323 @@ local t = require('luatest')
 local treegen = require('luatest.treegen')
 local justrun = require('luatest.justrun')
 
-local g = t.group()
+local tap_module_output_tests = {
+    {
+        test = [[test:ok(true)]],
+        pattern = 'ok - nil',
+    },
+    {
+        test = [[test:ok(true, 'test desc is ok')]],
+        pattern = 'ok - test desc is ok',
+    },
+    {
+        test = [[test:ok(false)]],
+        pattern = 'ok - nil',
+    },
+    {
+        test = [[test:ok(false, 'ok', { details = 'details', state = 'state'})]],
+        pattern = 'details: details',
+    },
+    {
+        test = [[test:ok(false, 'test desc is ok')]],
+        pattern = 'not ok - test desc is ok',
+    },
+    {
+        test = [[test:fail('failed test')]],
+        pattern = 'not ok - failed test',
+    },
+    {
+        test = [[test:skip('skipped test')]],
+        pattern = 'ok - skipped test # skip',
+    },
+    {
+        test = [[test:diag('diag message')]],
+        pattern = '# diag message',
+    },
+    {
+        test = [[test:is_deeply({a = 1}, {a = 1}, 'tables are equal')]],
+        pattern = 'ok - tables are equal',
+    },
+    {
+        test = [[test:is_deeply({a = 1}, {b = 1}, 'tables are not equal')]],
+        pattern = 'not ok - tables are not equal',
+    },
+    {
+        test = [[test:is(tonumber('48'), 48, 'tonumber(48) is 48')]],
+        pattern = 'ok - tonumber(48) is 48',
+    },
+    {
+        test = [[test:isnt(0xff, 64, '0xff is not 64')]],
+        pattern = 'ok - 0xff is not 64',
+    },
+    {
+        test = [[test:isnt(1, 1, '1 is not 1')]],
+        pattern = 'not ok - 1 is not 1',
+    },
+    {
+        test = [[test:isnil(nil, 'nil is nil')]],
+        pattern = 'ok - nil is nil',
+    },
+    {
+        test = [[test:isnil(48, '48 is nil')]],
+        pattern = 'not ok - 48 is nil',
+    },
+    {
+        test = [[test:isnumber(10, '10 is a number')]],
+        pattern = 'ok - 10 is a number',
+    },
+    {
+        test = [[test:isnumber(0, '0 is also a number')]],
+        pattern = 'ok - 0 is also a number',
+    },
+    {
+        test = [[test:isstring("blabla", '"blabla" is string')]],
+        pattern = 'ok - \"blabla\" is string',
+    },
+    {
+        test = [[test:isstring(48, '48 is string')]],
+        pattern = 'not ok - 48 is string',
+    },
+    {
+        test = [[test:isstring(nil, 'nil is string')]],
+        pattern = 'not ok - nil is string',
+    },
+    {
+        test = [[test:isboolean(true, 'true is boolean')]],
+        pattern = 'ok - true is boolean',
+    },
+    {
+        test = [[test:isboolean(1, '1 is boolean')]],
+        pattern = 'ok - 1 is boolean',
+    },
+    {
+        test = [[test:istable({}, '{} is a table')]],
+        pattern = 'ok - {} is a table',
+    },
+    {
+        test = [[test:isudata(nil, 'fiber', 'udata')]],
+        pattern = 'not ok - udata',
+    },
+    {
+        test = [[test:isudata(require('fiber').self(), 'some utype', 'udata')]],
+        pattern = 'ok - udata',
+    },
+    {
+        test = [[test:isudata(require('fiber').self(), 'fiber', 'udata')]],
+        pattern = 'ok - udata',
+    },
+    {
+        test = [[test:iscdata('xx', 'int', 'cdata type')]],
+        pattern = 'not ok - cdata type',
+    },
+    {
+        test = [[test:iscdata(10, 'int', 'cdata type')]],
+        pattern = 'not ok - cdata type',
+    },
+    {
+        test = [[test:iscdata(require('ffi').new('int', 10), 'int', 'cdata type')]],
+        pattern = 'ok - cdata type',
+    },
+    {
+        test = [[test:iscdata(require('ffi').new('unsigned int', 10), 'int', 'cdata type')]],
+        pattern = '',
+    },
+    {
+        test = [[test:like('abcde', 'cd', 'like(abcde, cd)')]],
+        pattern = 'ok - like(abcde, cd)',
+    },
+    {
+        test = [[test:unlike('abcde', 'acd', 'unlike(abcde, acd)')]],
+        pattern = 'ok - unlike(abcde, acd)',
+    },
+    -- gh-4125: strict nulls comparisons.
+    {
+        test = [[test:is(box.NULL, nil, "box.NULL == nil")]],
+        pattern = 'ok - box.NULL == nil',
+    },
+    {
+        test = [[test:is(nil, box.NULL, "nil == box.NULL")]],
+        pattern = 'ok - nil == box.NULL',
+    },
+    {
+        test = [[test:is(box.NULL, box.NULL, "box.NULL == box.NULL")]],
+        pattern = 'ok - box.NULL == box.NULL',
+    },
+    {
+        test = [[test:is(nil, nil, "nil == nil")]],
+        pattern = 'ok - nil == nil',
+    },
+    {
+        test = [[test:isnt(box.NULL, nil, "box.NULL != nil")]],
+        pattern = 'ok - box.NULL != nil',
+    },
+    {
+        test = [[test:isnt(nil, box.NULL, "nil != box.NULL")]],
+        pattern = 'ok - nil != box.NULL',
+    },
+    {
+        test = [[test:isnt(box.NULL, box.NULL, "box.NULL != box.NULL")]],
+        pattern = 'ok - box.NULL != box.NULL',
+    },
+    {
+        test = [[test:isnt(nil, nil, "nil != nil strict = true")]],
+        pattern = 'not ok - nil != nil strict = true',
+    },
+}
 
--- The testcase tests `ok`, `fail` and `skip` predicates.
-g.test_tap_module_test_statuses = function()
-    local dir = treegen.prepare_directory({}, {})
-    local script_name = 'tap_test_tap_module_test_statuses.lua'
-    treegen.write_file(dir, script_name, [[
-local tap = require('tap')
-local test = tap.test('root test')
-test.trace = false
-test:plan(5)
-test:ok(true, 'true') -- basic function
-local extra = { state = 'some useful information to debug on failure',
-        details = 'a table argument formatted using yaml.encode()' }
-test:ok(true, "extra information is not printed on success", extra)
-test:ok(false, "extra printed using yaml only on failure", extra)
-
-test:fail('failed') -- always fail the test
-test:skip('test marked as ok and skipped')
-    ]])
-    local opts = {nojson = true}
-    local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
-    t.assert_equals(res.stdout, [[
-TAP version 13
-1..5
-ok - true
-ok - extra information is not printed on success
-not ok - extra printed using yaml only on failure
-  ---
-  state: some useful information to debug on failure
-  details: a table argument formatted using yaml.encode()
-  ...
-not ok - failed
-ok - test marked as ok and skipped # skip]])
-end
+local pg = t.group('tap_output', tap_module_output_tests)
 
--- The testcase tests `is` and `isnt` predicates.
-g.test_tap_module_is_and_isnt = function()
+pg.test_tap_module_output = function(pg)
     local dir = treegen.prepare_directory({}, {})
-    local script_name = 'tap_test_tap_module_is_and_isnt.lua'
-    treegen.write_file(dir, script_name, [[
+    local script_name = 'tap_test_tap_module_output.lua'
+    local test_template = [[
 local tap = require('tap')
-local test = tap.test('root test')
+local test = tap.test('test name')
 test.trace = false
-test:plan(3)
-test:is(tonumber('48'), 48, 'tonumber(48) is 48')
-test:isnt(0xff, 64, '0xff is not 64')
-test:isnt(1, 1, '1 is not 1')
-    ]])
+test:plan(1)
+%s
+]]
+    local chunk = (test_template):format(pg.params.test)
+    treegen.write_file(dir, script_name, chunk)
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
-    t.assert_equals(res.stdout, [[
-TAP version 13
-1..3
-ok - tonumber(48) is 48
-ok - 0xff is not 64
-not ok - 1 is not 1
-  ---
-  unexpected: 1
-  got: 1
-  ...]])
+    t.assert_str_contains(res.stdout, 'TAP version 13')
+    t.assert_str_contains(res.stdout, '1..1')
+    t.assert_str_contains(res.stdout, pg.params.pattern)
 end
 
--- The testcase tests type predicates.
-g.test_tap_module_type_predicates = function()
-    local dir = treegen.prepare_directory({}, {})
-    local script_name = 'tap_test_tap_module_type_predicates.lua'
-    treegen.write_file(dir, script_name, [[
-local tap = require('tap')
-local test = tap.test('root test')
-test.trace = false
-test:plan(18)
-test:isnil(nil, 'nil is nil')
-test:isnil(48, '48 is nil')
-test:isnumber(10, '10 is a number')
-test:isnumber(0, '0 is also a number')
-test:isstring("blabla", '"blabla" is string')
-test:isstring(48, '48 is string')
-test:isstring(nil, 'nil is string')
-test:isboolean(true, 'true is boolean')
-test:isboolean(1, '1 is boolean')
-test:istable({}, '{} is a table')
-local udata = require('fiber').self()
-test:isudata(nil, 'fiber', 'udata')
-test:isudata(udata, 'some utype', 'udata')
-test:isudata(udata, 'fiber', 'udata')
-local ffi = require('ffi')
-test:iscdata('xx', 'int', 'cdata type')
-test:iscdata(10, 'int', 'cdata type')
-test:iscdata(ffi.new('int', 10), 'int', 'cdata type')
-test:iscdata(ffi.new('unsigned int', 10), 'int', 'cdata type')
-    ]])
-    local opts = {nojson = true}
-    local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
-    t.assert_equals(res.stdout, [[
-TAP version 13
-1..18
-ok - nil is nil
-not ok - 48 is nil
-  ---
-  expected: nil
-  got: 48
-  ...
-ok - 10 is a number
-ok - 0 is also a number
-ok - "blabla" is string
-not ok - 48 is string
-  ---
-  expected: string
-  got: number
-  ...
-not ok - nil is string
-  ---
-  expected: string
-  got: nil
-  ...
-ok - true is boolean
-not ok - 1 is boolean
-  ---
-  expected: boolean
-  got: number
-  ...
-ok - {} is a table
-not ok - udata
-  ---
-  expected: userdata<fiber>
-  got: nil
-  ...
-not ok - udata
-  ---
-  expected: userdata<some utype>
-  got: userdata<fiber>
-  ...
-ok - udata
-not ok - cdata type
-  ---
-  expected: ctype<int>
-  got: string
-  ...
-not ok - cdata type
-  ---
-  expected: ctype<int>
-  got: number
-  ...
-ok - cdata type
-not ok - cdata type
-  ---
-  expected: ctype<int>
-  got: ctype<unsigned int>
-  ...]])
-end
+local is_deeply_tests = {
+    {
+        got = {a = box.NULL},
+        expected = {b = box.NULL, c = box.NULL},
+        res_bool = true,
+    },
+    {
+        got = {a = box.NULL},
+        expected = {a = box.NULL},
+        res_bool = true,
+    },
+    {
+        got = {a = box.NULL},
+        expected = {b = box.NULL},
+        res_bool = true,
+    },
+    {
+        got = {a = box.NULL},
+        expected = {b = box.NULL, c = box.NULL},
+        res_bool = true,
+    },
+    {
+        got = {1, 2, 3},
+        expected = '200',
+        res_bool = false,
+    },
+    {
+        got = '200',
+        expected = {1, 2, 3},
+        res_bool = false,
+    },
+    {
+        got = 1,
+        expected = 1,
+        res_bool = true,
+    },
+    {
+        got = 'abc',
+        expected = 'abc',
+        res_bool = true,
+    },
+    {
+        got = {},
+        expected = {},
+        res_bool = true,
+    },
+    {
+        got = {1},
+        expected = {1},
+        res_bool = true,
+    },
+    {
+        got = {1},
+        expected = {2},
+        res_bool = false,
+    },
+    {
+        got = {1, 2, { 3, 4 }},
+        expected = {1, 2, { 3, 5 }},
+        res_bool = false,
+    },
+    -- gh-4125: `is_deeply` inconsistently works with `box.NULL`.
+    {
+        got = {},
+        expected = {a = box.NULL},
+        res_bool = true,
+    },
+    {
+        got = {a = box.NULL},
+        expected = {},
+        res_bool = true,
+    },
+    {
+        got = {a = box.NULL},
+        expected = {b = box.NULL},
+        res_bool = true,
+    },
+    {
+        got = nil,
+        expected = box.NULL,
+        res_bool = true,
+    },
+    {
+        got = box.NULL,
+        expected = nil,
+        res_bool = true,
+    },
+    {
+        got = {},
+        expected = {a = box.NULL},
+        res_bool = true,
+    },
+    {
+        got = {a = box.NULL},
+        expected = {},
+        res_bool = true,
+    },
+    {
+        got = nil,
+        expected = box.NULL,
+        res_bool = true,
+    },
+    {
+        got = box.NULL,
+        expected = nil,
+        res_bool = true,
+    },
+    {
+        got = {},
+        expected = {a = box.NULL},
+        res_bool = true,
+    },
+    {
+        got = nil,
+        expected = box.NULL,
+        res_bool = true,
+    },
+}
 
--- gh-4125: Strict nulls comparisons.
-g.test_tap_module_strict_null_comparisons = function()
-    local dir = treegen.prepare_directory({}, {})
-    local script_name = 'tap_test_tap_module_strict_null_comparisons.lua'
-    treegen.write_file(dir, script_name, [[
-local tap = require('tap')
-local test = tap.test('root test')
-test.trace = false
-test:plan(8)
-test.strict = true
-test:is(box.NULL, nil, "box.NULL == nil strict = true")
-test:is(nil, box.NULL, "nil == box.NULL strict = true")
-test:is(box.NULL, box.NULL, "box.NULL == box.NULL strict = true")
-test:is(nil, nil, "nil == nil strict = true")
-test:isnt(box.NULL, nil, "box.NULL != nil strict = true")
-test:isnt(nil, box.NULL, "nil != box.NULL strict = true")
-test:isnt(box.NULL, box.NULL, "box.NULL != box.NULL strict = true")
-test:isnt(nil, nil, "nil != nil strict = true")
-test.strict = false
-    ]])
-    local opts = {nojson = true}
-    local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
-    t.assert_equals(res.stdout, [[
-TAP version 13
-1..8
-not ok - box.NULL == nil strict = true
-  ---
-  got: null
-  ...
-not ok - nil == box.NULL strict = true
-  ---
-  expected: null
-  ...
-ok - box.NULL == box.NULL strict = true
-ok - nil == nil strict = true
-ok - box.NULL != nil strict = true
-ok - nil != box.NULL strict = true
-not ok - box.NULL != box.NULL strict = true
-  ---
-  unexpected: null
-  got: null
-  ...
-not ok - nil != nil strict = true]])
+local pg_is_deeply = t.group('tap_is_deeply', is_deeply_tests)
+
+pg_is_deeply.test_tap_is_deeply = function(pg)
+    local tap = require('tap')
+    local test = tap.test('test_tap_is_deeply')
+    test.trace = false
+    test:plan(1)
+    local res = test:is_deeply(pg.params.got, pg.params.expected)
+    t.assert_equals(res, pg.params.res_bool)
 end
 
+local g = t.group('tap')
+
 -- Any test also can create unlimited number of sub tests.
 -- Subtest with callbacks (preferred).
 -- Each subtest is counted in parent.
@@ -258,13 +374,9 @@ sub2:check() -- please call check() explicitly
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
-    t.assert_equals(res.stdout, [[
-TAP version 13
-1..8
-    1..1
-    ok - true in subtest
-    # hello from subtest
-ok - subtest 2]])
+    t.assert_str_contains(res.stdout, 'ok - true in subtest')
+    t.assert_str_contains(res.stdout, '# hello from subtest')
+    t.assert_str_contains(res.stdout, 'ok - subtest 2')
 end
 
 -- The testcase tests a TAP multisubtest.
@@ -382,252 +494,3 @@ not ok - failed subtests
   failed: 1
   ...]])
 end
-
-g.test_tap_module_is_deeply = function()
-    local dir = treegen.prepare_directory({}, {})
-    local script_name = 'tap_test_tap_module_is_deeply.lua'
-    treegen.write_file(dir, script_name, [[
-local tap = require('tap')
-local test = tap.test('root test')
-test.trace = false
-test:plan(8)
-test:test('is_deeply', function(t)
-    t:plan(21)
-
-    t:is_deeply(1, 1, '1 and 1')
-    t:is_deeply('abc', 'abc', 'abc and abc')
-    t:is_deeply({}, {}, 'empty tables')
-    t:is_deeply({1}, {1}, '{1} and {1}')
-    t:is_deeply({1}, {2}, '{1} and {2}')
-    t:is_deeply({1, 2, { 3, 4 }}, {1, 2, { 3, 5 }},
-                '{1,2,{3,4}} and {1,2,{3,5}}')
-
-    --
-    -- gh-4125: is_deeply inconsistently works with box.NULL.
-    --
-    t:is_deeply({}, {a = box.NULL}, '{} and {a = box.NULL} strict = false')
-    t:is_deeply({a = box.NULL}, {}, '{a = box.NULL} and {} strict = false')
-    t:is_deeply({a = box.NULL}, {b = box.NULL},
-                '{a = box.NULL} and {b = box.NULL} strict = false')
-    t:is_deeply({a = box.NULL}, {b = box.NULL, c = box.NULL},
-                '{a = box.NULL} and {b = box.NULL, c = box.NULL}\
-                 strict = false')
-    t:is_deeply(nil, box.NULL, 'nil and box.NULL strict = false')
-    t:is_deeply(box.NULL, nil, 'box.NULL and nil strict = false')
-    t:is_deeply({a = box.NULL}, {a = box.NULL},
-                '{a = box.NULL} and {a = box.NULL} strict false')
-
-    t.strict = true
-    t:is_deeply({}, {a = box.NULL}, '{} and {a = box.NULL} strict = true')
-    t:is_deeply({a = box.NULL}, {}, '{a = box.NULL} and {} strict = true')
-    t:is_deeply({a = box.NULL}, {b = box.NULL},
-                '{a = box.NULL} and {b = box.NULL} strict = true')
-    t:is_deeply({a = box.NULL}, {b = box.NULL, c = box.NULL},
-                '{a = box.NULL} and {b = box.NULL, c = box.NULL} strict = true')
-    t:is_deeply(nil, box.NULL, 'nil and box.NULL strict = true')
-    t:is_deeply(box.NULL, nil, 'box.NULL and nil strict = true')
-    t:is_deeply({a = box.NULL}, {a = box.NULL},
-                '{a = box.NULL} and {a = box.NULL} strict true')
-
-    t:test('check strict flag inheritance', function(t)
-        t:plan(2)
-        t:is_deeply({}, {a = box.NULL}, '{} and {a = box.NULL} strict = true')
-        t:is_deeply(nil, box.NULL, 'nil and box.NULL strict = true')
-    end)
-
-    t.strict = false
-end)
-    ]])
-    local opts = {nojson = true}
-    local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
-    t.assert_equals(res.stdout, [[
-TAP version 13
-1..8
-    # is_deeply
-    1..21
-    ok - 1 and 1
-    ok - abc and abc
-    ok - empty tables
-    ok - {1} and {1}
-    not ok - {1} and {2}
-      ---
-      path: //1
-      strict: false
-      expected: 2
-      got: 1
-      ...
-    not ok - {1,2,{3,4}} and {1,2,{3,5}}
-      ---
-      path: //3/2
-      strict: false
-      expected: 5
-      got: 4
-      ...
-    ok - {} and {a = box.NULL} strict = false
-    ok - {a = box.NULL} and {} strict = false
-    ok - {a = box.NULL} and {b = box.NULL} strict = false
-    ok - {a = box.NULL} and {b = box.NULL, c = box.NULL}
-                 strict = false
-    ok - nil and box.NULL strict = false
-    ok - box.NULL and nil strict = false
-    ok - {a = box.NULL} and {a = box.NULL} strict false
-    not ok - {} and {a = box.NULL} strict = true
-      ---
-      strict: true
-      expected: key a
-      got: nil
-      ...
-    not ok - {a = box.NULL} and {} strict = true
-      ---
-      path: //a
-      strict: true
-      expected: nil
-      got: cdata
-      ...
-    not ok - {a = box.NULL} and {b = box.NULL} strict = true
-      ---
-      path: //a
-      strict: true
-      expected: nil
-      got: cdata
-      ...
-    not ok - {a = box.NULL} and {b = box.NULL, c = box.NULL} strict = true
-      ---
-      path: //a
-      strict: true
-      expected: nil
-      got: cdata
-      ...
-    not ok - nil and box.NULL strict = true
-      ---
-      got: nil
-      expected: cdata
-      strict: true
-      ...
-    not ok - box.NULL and nil strict = true
-      ---
-      got: cdata
-      expected: nil
-      strict: true
-      ...
-    ok - {a = box.NULL} and {a = box.NULL} strict true
-        # check strict flag inheritance
-        1..2
-        not ok - {} and {a = box.NULL} strict = true
-          ---
-          strict: true
-          expected: key a
-          got: nil
-          ...
-        not ok - nil and box.NULL strict = true
-          ---
-          got: nil
-          expected: cdata
-          strict: true
-          ...
-        # check strict flag inheritance: end
-    not ok - failed subtests
-      ---
-      planned: 2
-      failed: 2
-      ...
-    # is_deeply: end
-not ok - failed subtests
-  ---
-  planned: 21
-  failed: 9
-  ...]])
-end
-
-g.test_tap_module_like_keyword = function()
-    local dir = treegen.prepare_directory({}, {})
-    local script_name = 'tap_test_tap_module_like_keyword.lua'
-    treegen.write_file(dir, script_name, [[
-local tap = require('tap')
-local test = tap.test('root test')
-test.trace = false
-test:plan(8)
-test:test('like', function(t)
-    t:plan(2)
-    t:like('abcde', 'cd', 'like(abcde, cd)')
-    t:unlike('abcde', 'acd', 'unlike(abcde, acd)')
-end)
-    ]])
-    local opts = {nojson = true}
-    local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
-    t.assert_equals(res.stdout, [[
-TAP version 13
-1..8
-    # like
-    1..2
-    ok - like(abcde, cd)
-    ok - unlike(abcde, acd)
-    # like: end
-ok - like]])
-end
-
--- Test, that in case of not strict comparison the order of
--- arguments does not matter.
-g.test_tap_module_comparison_order_does_not_matter = function()
-    local dir = treegen.prepare_directory({}, {})
-    local script_name =
-        'tap_test_tap_module_comparison_order_does_not_matter.lua'
-    treegen.write_file(dir, script_name, [[
-local tap = require('tap')
-local test = tap.test('root test')
-test.trace = false
-test:plan(2)
-test:is_deeply({1, 2, 3}, '200', "compare {1, 2, 3} and '200'")
-test:is_deeply('200', {1, 2, 3}, "compare '200' and {1, 2, 3}")
-    ]])
-    local opts = {nojson = true}
-    local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
-    t.assert_equals(res.stdout, [[
-TAP version 13
-1..2
-not ok - compare {1, 2, 3} and '200'
-  ---
-  strict: false
-  expected: '200'
-  got:
-  - 1
-  - 2
-  - 3
-  ...
-not ok - compare '200' and {1, 2, 3}
-  ---
-  strict: false
-  expected:
-  - 1
-  - 2
-  - 3
-  got: '200'
-  ...]])
-end
-
--- Finish root test. Since we used non-callback variant, we have
--- to call check explicitly.
-g.test_tap_module_check = function()
-    local dir = treegen.prepare_directory({}, {})
-    local script_name = 'tap_test_tap_module_check.lua'
-    treegen.write_file(dir, script_name, [[
-local tap = require('tap')
-local test = tap.test('root test')
-test.trace = false
-test:plan(2)
-test:ok(true)
-test:ok(true)
-test:check() -- call check() explicitly.
-    ]])
-    local opts = {nojson = true}
-    local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
-    t.assert_equals(res.stdout, [[
-TAP version 13
-1..2
-ok - nil
-ok - nil]])
-end
```